### PR TITLE
refactor!: extract httpc implementation to external library

### PR DESCRIPTION
**Pull request by kcloutie**
- replace inline HTTP client, caching, circuit breaker, compression, SSRF, and metrics transport with upstream oakwood-commons/httpc v0.1.0 library
- reduce pkg/httpc to a thin adapter that injects scafctl defaults (XDG cache dir, "scafctl:" key prefix, OTel metrics bridge)
- add OTelMetrics adapter bridging upstream Metrics interface to pkg/metrics OTel instruments
- add NewClientFromAppConfig bridge converting string-based config.HTTPClientConfig to typed ClientConfig
- preserve PrivateIPsAllowed(ctx) for application-layer SSRF checks; disable upstream transport-level SSRF gating to avoid double-check
- re-export all upstream types and sentinels so consumers need no import changes
- move ivan.dev/httpcache from direct to indirect dependency

BREAKING CHANGE: MergeHTTPClientConfig, wrapCheckRetryWithMetrics, and all cache/circuit-breaker/compression internal types are removed. NewClient now accepts *ClientConfig (pointer) instead of ClientConfig (value), and nil is allowed for scafctl defaults. SSRF error messages use capitalized "AllowPrivateIPs" field name.

### CODE REVIEW

File: pkg/httpc/httpc.go [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+	}
+	if local.Metrics == nil {
+		local.Metrics = &OTelMetrics{}
+	}
+	if local.CacheDir == "" {
+		local.CacheDir = paths.HTTPCacheDir()
+	}
+	if local.CacheKeyPrefix == "" {
+		local.CacheKeyPrefix = settings.DefaultHTTPCacheKeyPrefix
+	}
```
[the remark this answers is not in the document]
Comment (kcloutie) [reply pkg/httpc/httpc.go:88]: Fixed -- `NewClient` now uses `settings.HTTPCacheKeyPrefixFor(paths.AppName())` instead of the hardcoded constant. Thanks!

File: pkg/httpc/httpc.go [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+}
+
+// DefaultConfig returns a ClientConfig with scafctl-specific defaults:
+// XDG-based cache directory, "scafctl:" cache key prefix, and OTel metrics adapter.
+func DefaultConfig() *ClientConfig {
+	cfg := upstream.DefaultConfig()
+	cfg.CacheDir = paths.HTTPCacheDir()
+	cfg.CacheKeyPrefix = settings.DefaultHTTPCacheKeyPrefix
+	cfg.Metrics = &OTelMetrics{}
+	return cfg
```
[the remark this answers is not in the document]
Comment (kcloutie) [reply pkg/httpc/httpc.go:146]: Fixed -- `DefaultConfig` now uses `settings.HTTPCacheKeyPrefixFor(paths.AppName())` to derive the prefix from the current app name. Thanks!

File: pkg/httpc/bridge.go [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+// parseDurationOr parses s as a time.Duration, returning fallback when s is
+// empty or unparseable. Parse failures are logged with the given field name.
+func parseDurationOr(s string, fallback time.Duration, logger logr.Logger, field string) time.Duration {
+	if s == "" {
+		return fallback
+	}
+	d, err := time.ParseDuration(s)
+	if err != nil {
+		logger.Error(err, "invalid duration, using default", "field", field, "value", s)
+		return fallback
```
[the remark this answers is not in the document]
Comment (kcloutie) [reply pkg/httpc/bridge.go:27]: Good catch, fixed. Added a nil-sink guard (`if logger.GetSink() == nil { logger = logr.Discard() }`) before the error log, consistent with the pattern in `preflight.go`. Also added a test with a zero-value logger to cover the path.

File: pkg/httpc/httpc.go [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+	}
+	if local.CacheKeyPrefix == "" {
+		local.CacheKeyPrefix = settings.HTTPCacheKeyPrefixFor(paths.AppName())
+	}
+	// scafctl performs SSRF validation at the application layer via
+	// PrivateIPsAllowed(ctx) and ValidateURLNotPrivate(url) before issuing
+	// requests. Disable the upstream transport-level check to avoid double-gating
+	// and to preserve the original context-aware behaviour.
+	local.AllowPrivateIPs = true
+	return upstream.NewClient(&local)
```
[the remark this answers is not in the document]
Comment (kcloutie) [reply pkg/httpc/httpc.go]: Fixed -- `NewClient` now overrides `CheckRedirect` on the underlying `retryClient.HTTPClient` with a context-aware `ssrfSafeRedirectPolicy` that calls `PrivateIPsAllowed(req.Context())` and `ValidateURLNotPrivate` on each redirect target. This prevents SSRF bypasses via open redirects (e.g. public URL → 302 → `169.254.169.254`) while still allowing private IPs when the request context permits it. Added 3 tests covering: blocked private redirect, allowed private redirect, and max redirect enforcement. Thanks!

File: pkg/httpc/README.md [hunk 5]
```diff
@@ -1,1337 +1,36 @@
-# HTTP Client Package
+# pkg/httpc -- HTTP Client Adapter
 
-A robust HTTP client for Go with built-in support for retries, timeouts, and caching.
+Thin adapter over [`github.com/oakwood-commons/httpc`](https://github.com/oakwood-commons/httpc) that adds scafctl-specific behaviour:
 
-## Features
+- **XDG cache directory** via `pkg/paths` instead of `os.UserCacheDir`
+- **`scafctl:` cache key prefix** via `pkg/settings`
```
[the remark this answers is not in the document]
Comment (kcloutie) [reply pkg/httpc/README.md]: Good catch, fixed. Updated the README to say "App-name-derived cache key prefix" with an example showing `scafctl:` as the default.
